gym/f110_gym/envs/base_classes.py

```diff
@@ -266,143 +293,169 @@ def update_pose(self, raw_steer, vel):
         # state is [x, y, steer_angle, vel, yaw_angle, yaw_rate, slip_angle]
 
         # steering delay
-        steer = 0.
+        steer = 0.0
         if self.steer_buffer.shape[0] < self.steer_buffer_size:
-            steer = 0.
+            steer = 0.0
             self.steer_buffer = np.append(raw_steer, self.steer_buffer)
         else:
             steer = self.steer_buffer[-1]
             self.steer_buffer = self.steer_buffer[:-1]
             self.steer_buffer = np.append(raw_steer, self.steer_buffer)
 
+        if self.control_input == "speed":
```

Comment:
What about creating an action abstraction that process the given action and returns acceleration and steering velocity?

This would allow:
- a cleaner extension to more actions (e.g., discrete, meta-action, etc.), instead of starting nesting if-then-else. 
- better separation between action processing and update_pose


For example, assuming update_pose is only responsible for stepping the dynamics model given acceleration and steering velocity,  we could change line `640` as:
```
# looping over agents  
for i, agent in enumerate(self.agents):
    # update each agent's pose
    accl, sv = self.action_type.act(action)    # action_type implements the conversion to accl, sv
    current_scan = agent.update_pose(accl, sv)
    agent_scans.append(current_scan)
```

Reply:
Added classes for action types and specific implementations when they take actions to accl and sv.
I kept the accl and sv calculation at where it is right now, and will change that in a new pr with the update_pose changes.